pootle/apps/pootle_store/diff.py

```diff
@@ -0,0 +1,226 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+from collections import OrderedDict
+import difflib
+
+from django.utils.functional import cached_property
+
+from .util import OBSOLETE
+from .fields import to_python as multistring_to_python
+
+
+class StoreDiff(object):
+
+    def __init__(self, db_store, file_store, file_revision):
+        self.db_store = db_store
+        self.file_store = file_store
+        self.db_revision = db_store.get_max_unit_revision()
+        self.file_revision = file_revision
+
+    def _active_units(self):
+        return [unitid for unitid, unit in self.db_units.items()
+                if unit['state'] != OBSOLETE]
+    active_units = cached_property(_active_units)
```

Comment:
because i had been trying to work out why cached_property wasnt working - ill revert